### DISCUSSION TITLE
Feedback about documentation/debugging

### DISCUSSION THREAD

**Discussion by Destroy666x** [JanDeDobbeleer/oh-my-posh#3055] [General]
I've been trying to set this up with latest ConEmu + lastest clink + latest version of this software. So far my experience isn't great in terms of documentation and this is my perspective as someone who likes to set up software from scratch.

I got lots of `oh-my-posh.lua:1: attempt to call a nil value` errors, which were all related to misconfiguration as it ran just fine without the `--config` parameter. It happened even when using some examples from documentation. E.g. this: https://ohmyposh.dev/docs/configuration/overview doesn't work by default. Apparently it's missing `version: 2` field, which interestingly is sometimes added to the file after the initialization, modifying the config file (and messing the order up), but not always. Instead of having the software attempt to add it if it's missing, I think the priority should be for the documentation to properly reflect that it's required.

I also saw that the recommended way of validating the config was following the JSON schema: https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json. I've followed all the `required` fields (also mostly not mentioned in the docs), added some `style`s for `segment`s but it was still wrong. Despite online YAML/JSON (after converting the YAML) validators claiming that the syntax is correct and the config matches the schema.

It turned out I mistakenly had an array item instead of one parameter field in my YAML in a segment. I had to use `oh-my-posh --config D:/Backups/Applications/Configs/General/Linkable/Oh-My-Posh/oh-my-posh.yml debug` command which also wasn't really mentioned in the documentation, in FAQ I found `debug` for some other question, but the param had to be thought of manually. I think this way should be recommended instead of or next to the JSON schema validation that doesn't catch everything.

Just sharing my thoughts about some things that could be improved. Otherwise - great software to be trying.

**Comment by JanDeDobbeleer** [comment]
> It turned out I mistakenly had an array item instead of one parameter field in my YAML in a segment

@Destroy666x which field was this?

**Reply by Destroy666x** [reply]
Just a random property (sorry, not parameter) of a segment, I don't remember which one exactly since I've had like 10 segment prepared to be tested at that time. It was the only property, otherwise I'm sure there'd be a syntax error of yaml.

So something like:
```
blocks:
  - type: prompt
    alignment: left
    segments:
      - type: path
        style: plain
        properties:
          - style: agnoster_full
```
EDIT: as I see they are in theory defined as object for all the segments in the schema, in a similar way, hmm. Not sure why the validator didn't throw an error then. Maybe that bugged out because of 3x nested `properties`.
